### PR TITLE
Fix error in console on rejecting signature request

### LINKED ISSUE #18550: [Bug]: eth_sign - Cannot destructure property 'currentLocale' of 'e' as it is undefined
### Describe the bug

This issue is NOT limited to eth_sign, it affects:
- Eth Sign
- Personal Sign
- Signed Type Data
- SIWE

https://user-images.githubusercontent.com/53189696/231540640-81f5a2f8-ac54-453e-aa35-fb9548231921.mov

### Steps to reproduce

1. Enable eth_sign
2. navigate to the test-dapp
3. click eth sign
4. do not accept/reject the confirmation
5. switch to MM extension
6. open the unconfirmed transaction from history
7. click reject 

### Error messages or log output

```shell
runtime-lavamoat.js:7100 SES_UNHANDLED_REJECTION: TypeError: Cannot destructure property 'currentLocale' of 'e' as it is undefined.
  at ui-10.js:1:52670
  at index.js:11:1
  at e.<computed> (sentry-install.js:1:682)
  at dispatch (redux.js:699:1)
  at actions.ts:1274:5
  at async onCancel (signature-request-original.component.js:246:19)
handleRejection @ runtime-lavamoat.js:7100
finalizeDroppedPromise @ runtime-lavamoat.js:6928
```

### Version

10.29

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

Ledger

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: #18550

Fix error in console on rejecting signature requests.

### CODE REVIEW

**Comment by Silvrbckw** [comment]
Merge

File: app/scripts/controllers/sign.ts [hunk 1]
```diff
@@ -414,7 +414,7 @@ export default class SignController extends BaseControllerV2<
    * @param msgId - The id of the message to cancel.
```
Comment (matthewwalsh0) [inline app/scripts/controllers/sign.ts:414]: For completion sake, should we add the return value to the method doc also so it's clear we return the full background state?

File: app/scripts/controllers/sign.ts [hunk 2]
```diff
@@ -414,7 +414,7 @@ export default class SignController extends BaseControllerV2<
...
    */
   cancelMessage(msgId: string) {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/scripts/controllers/sign.ts:417]: Good point @matthewwalsh0 , I updated the PR.

**Comment by seaona** [comment]
@jpuri from QA looks good! I see the issue fixed for all signature types
- eth_sign
- personal_sign
- sign_typeddata
- sign_typeddata_v3
- sign_typeddata_v4

[Screencast from 17-04-23 11:26:46.webm](https://user-images.githubusercontent.com/54408225/232443616-f9429050-a25b-4ad5-9e14-e1b2513376c0.webm)

**Review by NiranjanaBinoy — APPROVED** [review]
LGTM

**Review by digiwand — APPROVED** [review]

**Pull request merged**